packages/flame/lib/src/effects/effects.dart

```diff
@@ -28,34 +35,95 @@ abstract class ComponentEffect<T extends BaseComponent> {
   bool isInfinite;
   bool isAlternating;
   final bool isRelative;
+  final bool removeOnFinish;
   final bool _initialIsInfinite;
   final bool _initialIsAlternating;
-  double? percentage;
+
+  /// The percentage of the effect that has passed including [preOffset] and
+  /// [postOffset].
+  double percentage = 0.0;
+
+  /// The outcome the curve function, only updates after [preOffset] and before
+  /// [postOffset]
   double curveProgress = 0.0;
+
+  /// How much time it takes for the effect to peak, which means right before it
+  /// starts any potential reversal or reset. Including both offsets.
   double peakTime = 0.0;
+
+  /// The time passed since the start of the effect, it will start to decrease
+  /// after it has reached [peakTime] if [isAlternating] is true, and reset to
+  /// zero if it is not.
   double currentTime = 0.0;
+
+  /// When an effect reaches the end, and the beginning if it is alternating,
+  /// it will overshoot 0.0 and [peakTime], this time is added to the next time
+  /// step.
   double driftTime = 0.0;
+
+  /// Whether the effect is going forward or is reversing.
+  ///
+  /// Reversing in this context means that after the effect has peaked and if it
+  /// has [isAlternating] set to true, it will do the effect backwards back to
+  /// its original state.
   int curveDirection = 1;
+
+  /// Which curve that the effect is following.
   Curve curve;
 
+  /// The time (in seconds) that should pass before the component starts each
+  /// iteration.
+  double preOffset;
```

Comment:
this is a totally unrelated addition right?

Reply:
I thought that it would be needed to make the `CombinedEffect` work like it used to work, we'll see if it is actually possible to make the `CombinedEffect` alternate with the current structure.